### PR TITLE
Add dependencies to themebuilder metadata

### PR DESCRIPTION
**Pull request by babich-a**

### CODE REVIEW

File: themebuilder/modules/metadata-generator.js [hunk 1]
```diff
@@ -4,66 +4,158 @@ const fs = require("fs");
...
+                process: (less, context) => {
+                    const fullPath = context && context.fileInfo && context.fileInfo.filename;
+
+                    if(typeof fullPath === "string") {
+                        const widget = getWidgetFromFileName(path.basename(fullPath));
+                        if(!dependencies[widget]) {
+                            dependencies[widget] = [];
+                        }
+
+                        dependencies[widget] = dependencies[widget].concat(parseImports(less));
```
Comment (BingoRUS) [inline themebuilder/modules/metadata-generator.js]:
```suggestion
                        dependencies[widget].push(...parseImports(less));
```

File: themebuilder/modules/metadata-generator.js [hunk 2]
```diff
@@ -4,66 +4,158 @@ const fs = require("fs");
...
+};
+
+const resolveDependencies = (dependencies) => {
+    for(var widget in dependencies) {
+        if(Object.prototype.hasOwnProperty.call(dependencies, widget)) {
+            const widgetDependencies = dependencies[widget];
+
+            for(let i = 0; i < widgetDependencies.length; i++) {
+                const dependencyWidget = widgetDependencies[i];
+                const dependencyList = dependencies[dependencyWidget];
```
Comment (BingoRUS) [inline themebuilder/modules/metadata-generator.js:67]: Shouldn't it be recursive?

File: themebuilder/modules/metadata-generator.js [hunk 3]
```diff
@@ -4,66 +4,158 @@ const fs = require("fs");
...
+            if(selfIndex >= 0) {
+                widgetDependencies.splice(selfIndex, 1);
+            }
+        }
+    }
+
+    return dependencies;
+};
+
+const removeDependenciesInternals = (dependencies) => {
```
Comment (BingoRUS) [inline themebuilder/modules/metadata-generator.js]: I think `removeInternalDependencies` is a better method name.

File: themebuilder/modules/metadata-generator.js (hunk 2 above)
Comment (babich-a) [reply themebuilder/modules/metadata-generator.js:67]: No. We add the dependency to the end of array and then in any case process it. Theoretically, the algorithm should recursive (it is a tree), but our tree has circle dependencies and recursion can cause the trouble :)

**Review by BingoRUS — APPROVED** [review]

**Pull request merged**